### PR TITLE
Plugins: Fix loading symbolically linked plugins

### PR DESCRIPTION
**Pull request by domasx2**
**What this PR does / why we need it**:

Hey!

It seems [this change](https://github.com/grafana/grafana/commit/83f26e9ce274d20e25b90fee0fc0907fc9623d27#diff-c3e51d4a9bdb7b3197a7d4dd09c9ac236c1add5a4be3c8bd64fd3bd52e2dad9cR59) broke loading plugins that are symbolically linked to plugins dir.
It intends to check if symbolic link links to a directory, but `.IsDir()` is always false for `os.Lstat`,  `os.Stat` should be used instead. This PR reproduces and fixes the issue

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/plugins/manager/manager_test.go [hunk 1]
```diff
@@ -397,6 +397,25 @@ func TestPluginManager_Init(t *testing.T) {
...
+		pm := createManager(t, func(pm *PluginManager) {
+			pm.Cfg.PluginsPath = "testdata/symbolic-plugin-dirs"
+		})
+		err := pm.Init()
+		require.NoError(t, err)
+		// This plugin should be properly registered, even though it is symlinked to prlugins dir
+		require.Empty(t, pm.scanningErrors)
+		const pluginID = "test"
+		assert.NotNil(t, pm.plugins[pluginID])
+		assert.NotNil(t, pm.plugins[(pluginID)])
```
Comment (wbrowne) [inline pkg/plugins/manager/manager_test.go:417]: Can remove this it looks like

**Review by wbrowne — APPROVED** [review]
Nice one @domasx2  - didn't even know this was supported/used this way.

LGTM assuming tests pass

File: pkg/plugins/manager/manager_test.go (hunk 1 above)
Comment (domasx2) [reply pkg/plugins/manager/manager_test.go]: done :)

**Comment by domasx2** [comment]
> Nice one @domasx2 - didn't even know this was supported/used this way.
I have the plugins I work on checked out to a separate directotory, and symlink them into grafana `data/plugins` dir. When grafana failed to start, first I was alarmed that linux updates I had just installed somehow broke symlinks :sweat_smile:

**Pull request merged**